static/app/views/alerts/rules/issue/messagingIntegrationModal.tsx

```diff
@@ -43,15 +81,37 @@ function MessagingIntegrationModal({
         {bodyContent}
         <IntegrationsWrapper>
           {providerKeys.map((value: string) => {
+            getProvider(value, provider, setProvider, organization, api, setHasError);
+            if (!provider) {
+              return null;
+            }
             return (
-              <AddIntegrationRow
-                key={value}
-                providerKey={value}
+              <Access
+                access={['org:integrations']}
                 organization={organization}
-                project={project}
-                onClickHandler={closeModal}
-                setHasError={setHasError}
-              />
+                key={value}
+              >
+                {({hasAccess}) => {
+                  return (
+                    <IntegrationContext.Provider
+                      value={{
+                        provider: provider,
```

Comment:
it seems like most of the data we would want is really in the `provider` key. things like `organization` should likely be passed as a property for now.

I'd recommend looking at this context and making sure it's just the API response information needed for this component. higher level things like organization and project should likely be refactored into their own providers later.